storage/grpc_client.go

```diff
@@ -996,8 +990,15 @@ func (c *grpcStorageClient) NewRangeReader(ctx context.Context, params *newRange
 		r.remain = size
 	}
 
+	// For a zero-length request, explicitly close the stream and set remaining
+	// bytes to zero.
+	if params.length == 0 {
```

Comment:
Do we want to consider improving perf for zero-length requests by requesting less bytes to start with as we are just going to discard them?

Reply:
I don't think this is enough of a mainstream use case to optimize for; also, only the first 2MiB of the object will come through in the first `recv()`. This seems fine to me.